my_game.py

```diff
@@ -155,18 +155,19 @@ def on_update(self, delta_time):
 
 class Asteroid(arcade.Sprite):
 
-    valid_sizes = {1: "images/Meteors/meteorGrey_tiny1.png", 2: "images/Meteors/meteorGrey_small1.png", 3: "images/Meteors/meteorGrey_med1.png", 4: "images/Meteors/meteorGrey_big1.png"}
-
+    valid_sizes = (1, 2, 3, 4)
+    graphics = {1: "images/Meteors/meteorGrey_tiny1.png", 2: "images/Meteors/meteorGrey_small1.png", 3: "images/Meteors/meteorGrey_med1.png", 4: "images/Meteors/meteorGrey_big1.png"}
+    
     def __init__(self, size=None):
 
         # Initialize the asteroid
         
         if size is None:
-            size = random.randrange(1, 4)
+            size = random.choice(Asteroid.valid_sizes)
```

Comment:
You don't need the list of sizes in _valid_sizes_. You can get the sizes from _graphics_ like this:
```
random.randrange(graphics.keys())
```

Reply:
I tried that but it turns out that the .keys() function returns this:
dict_keys(<key1>, <key2>)

Reply:
i would have to do something like this
        if size is None:
            keys = []
            for k in Asteroid.valid_sizes.keys()
                keys.append(k)
            size = random.choice(keys)